cmake/CompilerHelper.cmake

```diff
@@ -10,3 +10,11 @@ if(CMAKE_C_COMPILER_ID MATCHES "Clang")
   set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS
       "${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS} -undefined dynamic_lookup")
 endif()
+
+if(CMAKE_C_COMPILER_ID MATCHES "PGI")
```

Comment:
In thinking about how to figure this out in autotools, I began to wonder if this is the right approach @pramodk . That is,
since it is only used in setup.py and in all the c/cxx files one directly uses __PGI . Wouldn't it be safest to establish within setup.py that the compiler it will be using does in fact have __PGI defined

Reply:
I committed in my version
```
$ git show HEAD
...
    Autotools build determines USING_PGI_COMPILER_TRUE from $CC
...
diff --git a/configure.ac b/configure.ac
...
+dnl Some things are specific to the PGI compiler
+AM_CONDITIONAL(USING_PGI_COMPILER, $CC --version | grep -q pgcc)
+
```
but I'm holding off on pushing it until I see that this is the only thing preventing a successful travis-ci. I can imagine many ways in which this is unsafe but it should work when one is NOT using pgi.